cmd/sops/common/common.go

```diff
@@ -36,6 +36,37 @@ type Store interface {
 	ExampleFileEmitter
 }
 
+type storeConstructor = func() Store
+
+func newBinaryStore() Store {
+	return &json.BinaryStore{}
+}
+
+func newDotenvStore() Store {
+	return &dotenv.Store{}
+}
+
+func newIniStore() Store {
+	return &ini.Store{}
+}
+
+func newJsonStore() Store {
+	return &json.Store{}
+}
+
+func newYamlStore() Store {
+	return &yaml.Store{}
+}
+
+// StoreConstructors is a mapping of formats to constructors.
+var StoreConstructors = map[string]storeConstructor{
+	"binary": newBinaryStore,
```

Comment:
I would prefer to store these as something similar to an enum:

```go
type Format int

const (
  Binary Format = iota
  Json
  Ini
  // ...
)
```

And

```go
var storeConstructors = map[Format]storeConstructor{
   Yaml: newYamlStore,
// ...
```

Reply:
the `sops` CLI uses strings, so i'm not sure what you want you want me to use there.

Reply:
The CLI should also convert from strings to `Format` ASAP